elixir/apps/domain/test/support/fixtures/subject_fixtures.ex

```diff
@@ -48,27 +49,55 @@ defmodule Domain.SubjectFixtures do
         }
       end)
 
-    token =
-      Map.get_lazy(attrs, :token, fn ->
-        token_attrs =
-          attrs
-          |> Map.take([:expires_at, :name])
-          |> Enum.into(%{
-            type: token_type,
-            account: account,
-            actor: actor
-          })
-
-        token_fixture(token_attrs)
-      end)
+    # Portal sessions for portal context, tokens for client/api_client
+    case context_type do
+      :portal ->
+        session =
+          Map.get_lazy(attrs, :session, fn ->
+            session_attrs =
+              attrs
+              |> Map.take([:expires_at])
+              |> Enum.into(%{
+                account: account,
+                actor: actor
+              })
+
+            portal_session_fixture(session_attrs)
+          end)
+
+        %Subject{
+          actor: actor,
+          account: account,
+          expires_at: session.expires_at,
+          context: context,
+          auth_ref: %{type: :portal_session, id: session.id},
+          auth_provider_id: session.auth_provider_id
+        }
 
-    %Subject{
-      actor: actor,
-      account: account,
-      expires_at: token.expires_at,
-      context: context,
-      token_id: token.id
-    }
+      type when type in [:client, :api_client] ->
+        token =
+          Map.get_lazy(attrs, :token, fn ->
+            token_attrs =
+              attrs
+              |> Map.take([:expires_at, :name])
+              |> Enum.into(%{
+                type: context_type,
+                account: account,
+                actor: actor
+              })
+
+            token_fixture(token_attrs)
+          end)
+
+        %Subject{
+          actor: actor,
+          account: account,
+          expires_at: token.expires_at,
+          context: context,
+          auth_ref: %{type: :token, id: token.id},
+          auth_provider_id: token.auth_provider_id
+        }
+    end
```

Comment:
Curious on your thoughts around potentially refactoring this case statement?

```suggestion
    {auth_type, auth_struct} =
      case context_type do
        :portal ->
          session =
            Map.get_lazy(attrs, :session, fn ->
              attrs
              |> Map.take([:expires_at])
              |> Enum.into(%{account: account, actor: actor})
              |> portal_session_fixture()
            end)
    
          {:portal_session, session}
    
        type when type in [:client, :api_client] ->
          token =
            Map.get_lazy(attrs, :token, fn ->
              attrs
              |> Map.take([:expires_at, :name])
              |> Enum.into(%{type: context_type, account: account, actor: actor})
              |> token_fixture()
            end)
    
          {:token, token}
      end
    
    %Subject{
      actor: actor,
      account: account,
      expires_at: auth_struct.expires_at,
      context: context,
      auth_ref: %{type: auth_type, id: auth_struct.id},
      auth_provider_id: auth_struct.auth_provider_id
    }
```

Reply:
Not a blocker, just something that came to mind when looking through the PR.

Reply:
Good catch, that pattern is a bit simpler, fixed.